### PR TITLE
postpone model import in `channels.auth` to avoid ImproperlyConfigured error

### PR DESCRIPTION
**Pull request by yedpodtrzitko**
Hi,
the import of `channels.auth` is triggering import of `AnonymousUser` model, which should not happen before ASGI app is ready. Doing so raises `ImproperlyConfigured` exception which might be tricky for newcomers to solve (see numerous Issues and a few MR trying to pinpoint this in docs, eg. #1681, #1691)

To make it work, some code needs to be mixed among imports (`django.setup()` or `get_asgi_application()`), which does not look nice.

Postponing the model import can solve the most common culprit of this error.

### CODE REVIEW

File: channels/auth.py [hunk 1]
```diff
@@ -24,6 +23,9 @@ def get_user(scope):
     Return the user model instance associated with the given scope.
     If no user is retrieved, return an instance of `AnonymousUser`.
     """
+    # postpone model import to avoid ImproperlyConfigured error before ASGI app is ready
```
Comment (carltongibson) [inline channels/auth.py:26]: `...before Django setup is complete`? Maybe.

**Review by carltongibson — COMMENTED** [review]
Hi @yedpodtrzitko — Thanks for this, yes, I think it's probably a good idea.

File: channels/auth.py (hunk 1 above)
Comment (yedpodtrzitko) [reply channels/auth.py]: done

**Comment by yedpodtrzitko** [comment]
I moved the docs changes from #1693 here, as it makes sense to have it merged atomically.

- since `channels.auth` can be imported early now, I removed the mention of `AuthMiddlewareStack` as a problematic code and moved its import above `get_asgi_application()`
- the code sample in `routing.rst` looks like it would likely lead to the `ImproperlyConfigured` error too, since it's calling `get_asgi_application()` very late, so I replaced it with the example from `deploying.rst` to make it more unified.
- fixed a few typos

**Comment by carltongibson** [comment]
Ok, thanks. I just need to have a little play. 👍

**Comment by yedpodtrzitko** [comment]
@carltongibson I assume you're super busy, but is there any chance merging this soon? This seems like the most common issue newcomers are hitting and it's spawning a lot of redundant Issues and MRs (eg. fresh #1707)...

**Comment by carltongibson** [comment]
Hi @yedpodtrzitko — yes, I'm just getting some bandwidth to swing back here now. This is on my agenda for the next point release. (Thanks! 👍)

**Review by carltongibson — APPROVED** [review]
Hi @yedpodtrzitko — sorry for the slow uptake here. This looks great. Thanks!

**Pull request merged**